File: src/typescript/key-generation/generate-key/suggest-brain-key.ts
import { createWaxFoundation } from '@hiveio/wax';

const waxApi = await createWaxFoundation();

// Suggest brain key
const privateKeyData = waxApi.suggestBrainKey();

console.log(`Associated Public Key: ${privateKeyData.associatedPublicKey}`);
console.log(`WIF Private Key: ${privateKeyData.wifPrivateKey}`);
console.log(`Brain Key: ${privateKeyData.brainKey}`);

File: src/rust/key_generation/generate_key/suggest_brain_key.rs
use wax::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let wax_api = create_wax_foundation(None);

    // Suggest brain key
    let private_key_data = wax_api.suggest_brain_key()?;

    println!(
        "Associated Public Key: {}",
        private_key_data.associated_public_key
    );
    println!("WIF Private Key: {}", private_key_data.wif_private_key);
    println!("Brain Key: {}", private_key_data.brain_key);

    Ok(())
}
